File: WebProject/src/model/Order.ts
import { OrderItem } from './OrderItem';
import {OrderState} from './OrderState';
export class Order{

    id: number = 0;
    deliveryAddress: string = '';
    purchaserId:number=-1;
    items: OrderItem[] = [];
    price: number = 0;
    state :OrderState=OrderState.created;
    created:Date=new Date()
    started:Date=new Date()
    ended:Date=new Date();
    deliveryTime:number=0;
    comment:String="";

    public ToString(orderItem: OrderItem) {
      return (
        'ID: ' +
        orderItem.id +
        'Name: ' +
        orderItem.name +
        'Quantity: '+
        orderItem.quantity+
        'Price: ' +
        orderItem.price
      );
    }
}

File: OnlineDeliveryServer/OnlineDeliveryServer/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OnlineDeliveryServer
{
    public class Order
    {
        public int Id { get; set; }
        public string DeliveryAddress { get; set; }
        public string Comment { get; set; }
        public double Price { get; set; }
        public OrderState State { get; set; }
        public DateTime Created { get; set; }
        public DateTime Started { get; set; }
        public DateTime Ended { get; set; }
        public double DeliveryTime { get; set; }
        public ICollection<OrderItem> Items { get; set; }
        public int DelivererId { get; set; }
        public User Purchaser { get; set; }
        public int PurchaserId { get; set; }
        public Order(int id, string deliveryAddress, string comment, double price)
        {
            Id = id;
            DeliveryAddress = deliveryAddress;
            Comment = comment;
            Price = price;
            this.Created=DateTime.Now;
            this.State = OrderState.created;
        }
        public Order() { }
    }
}
